business_requirement/models/business.py

```diff
@@ -0,0 +1,446 @@
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class BusinessRequirement(models.Model):
+    _inherit = 'mail.thread'
+    _name = "business.requirement"
+    _description = "Business Requirement"
+    _order = 'name desc'
+
+    @api.model
+    def _get_default_company(self):
+        if not self.env.user.company_id:
+            raise ValidationError(
+                _('There is no default company for the current user!'))
+        return self.env.user.company_id.id
+
+    sequence = fields.Char(
+        'Sequence',
+        readonly=True,
+        copy=False,
+        index=True,
+    )
+    name = fields.Char(
+        'Name',
+        required=False,
+        readonly=True,
+        copy=False,
+        states={'draft': [('readonly', False)]}
+    )
+    description = fields.Char(
+        'Description', required=True,
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    ref = fields.Char(
+        'WBS',
+        required=False,
+        readonly=True,
+        copy=False,
+        states={'draft': [('readonly', False)]}
+    )
+    business_requirement = fields.Html(
+        'Customer Story',
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    scenario = fields.Html(
+        'Scenario',
+        readonly=True,
+        states={'draft': [('readonly', False)],
+                'confirmed': [('readonly', False)]}
+    )
+    gap = fields.Html(
+        'Gap',
+        readonly=True,
+        states={'draft': [('readonly', False)],
+                'confirmed': [('readonly', False)]}
+    )
+    test_case = fields.Html(
+        'Test Case',
+        readonly=True,
+        states={'draft': [('readonly', False)],
+                'confirmed': [('readonly', False)]}
+    )
+    terms_and_conditions = fields.Html(
+        'Terms and Conditions',
+        readonly=True,
+        states={'draft': [('readonly', False)],
+                'confirmed': [('readonly', False)]}
+    )
+    category_ids = fields.Many2many(
+        'business.requirement.category',
+        string='Categories',
+        relation='business_requirement_category_rel',
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    state = fields.Selection(
+        selection="_get_states",
+        string='State',
+        default='draft',
+        copy=False,
+        readonly=False,
+        states={'draft': [('readonly', False)]},
+        track_visibility='onchange'
+    )
+    business_requirement_ids = fields.One2many(
+        comodel_name='business.requirement',
+        inverse_name='parent_id',
+        string='Sub Business Requirement',
+        copy=False,
+        readonly=True,
+        states={'draft': [('readonly', False)]}
+    )
+    parent_id = fields.Many2one(
+        comodel_name='business.requirement',
+        string='Parent',
+        ondelete='set null',
+        domain="[('id', '!=', id)]",
```

Comment:
If I remember this was to prevent having a parent business_requirement which it's parent is itself.
But as I will delete sub BR this field will disappear as well